custom_components/mail_and_packages/camera.py

```diff
@@ -230,10 +270,231 @@ def update_file_path(self) -> None:
+                        )
+                    else:
+                        _LOGGER.warning(
+                            "Failed to create animated GIF, using first delivery image"
+                        )
+                        file_path = delivery_images[0]
+
+                elif len(delivery_images) == 1:
+                    # Single delivery image
+                    file_path = delivery_images[0]
+                    _LOGGER.debug(
+                        "Generic camera - using single delivery image: %s", file_path
+                    )
+
+                else:
+                    # No deliveries found, use default generic no mail image
+                    _LOGGER.debug(
+                        "Generic camera - no deliveries found, using default: %s",
+                        file_path,
+                    )
+
         self.check_file_path_access(file_path)
         self._file_path = file_path
         self.schedule_update_ha_state()
 
+    def _is_custom_no_mail_image(self, base_name: str, file_path: str) -> bool:
+        """Check if the given file path is a custom 'no mail' image for the specified camera.
+
+        Args:
+            base_name: The base name of the camera (e.g., 'amazon', 'ups', 'walmart', 'usps')
+            file_path: The full file path to check
+
+        Returns:
+            True if this is a custom 'no mail' image, False otherwise
+        """
+        # Handle USPS camera differently (uses CONF_CUSTOM_IMG instead of CONF_USPS_CUSTOM_IMG)
+        if base_name == "usps":
+            custom_img_key = "CONF_CUSTOM_IMG"
+            custom_img_file_key = "CONF_CUSTOM_IMG_FILE"
+        else:
+            # Handle other cameras (Amazon, UPS, Walmart)
+            custom_img_key = f"CONF_{base_name.upper()}_CUSTOM_IMG"
+            custom_img_file_key = f"CONF_{base_name.upper()}_CUSTOM_IMG_FILE"
+
+        custom_img_conf = getattr(const, custom_img_key, None)
+        custom_img_file_conf = getattr(const, custom_img_file_key, None)
+
+        if (
+            custom_img_conf
+            and custom_img_file_conf
+            and self.config.data.get(custom_img_conf)
+        ):
+            custom_file_path = self.config.data.get(custom_img_file_conf)
+            if custom_file_path and os.path.exists(custom_file_path):
+                # Check if the file path matches the custom "no mail" image
+                return os.path.abspath(file_path) == os.path.abspath(custom_file_path)
+
+        return False
+
+    def _get_sensor_name_for_camera(self, camera_type: str) -> str:
+        """Get the sensor name that corresponds to a camera type.
+
+        Args:
+            camera_type: The camera type (e.g., 'amazon_camera', 'ups_camera', etc.)
+
+        Returns:
+            The corresponding sensor name, or None if no mapping exists
+        """
+        # Extract base name from camera type (e.g., "amazon_camera" -> "amazon")
+        base_name = camera_type.replace("_camera", "")
```

Comment:
```suggestion
        base_name = camera_type.split("_")[0]
```

We can just use a split here rather than a replace.